Lib/test/test_socket.py

```diff
@@ -6826,6 +6826,31 @@ class TestMacOSTCPFlags(unittest.TestCase):
     def test_tcp_keepalive(self):
         self.assertTrue(socket.TCP_KEEPALIVE)
 
+@unittest.skipUnless(hasattr(socket, 'TCP_QUICKACK'), 'need socket.TCP_QUICKACK')
+class TestQuickackFlag(unittest.TestCase):
+    def check_set_quickack(self, sock):
+        # quickack already true by default on some OS distributions
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        if opt:
+            sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK, 0)
+
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        self.assertFalse(opt)
+
+        sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK, 1)
+
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        self.assertTrue(opt)
+
+    def test_set_quickack(self):
+        for blocking in (False, True):
+            with self.subTest(blocking=blocking):
```

Comment:
Is it useful to test a non-blocking socket? Blocking or not sounds unrelated to QUICKACK option.

Reply:
I just copied the TCP_NODELAY test (pretty sure it was that one), not sure why it's testing a blocking socket either but figured it couldn't hurt (shrug).  I can remove it if you want.

Reply:
I suggest to remove the non-blocking test, just test a regular (blocking) socket. I prefer to start with simple test.

Reply:
no worries, will do so now